pkgs/development/python-modules/numba/default.nix

```diff
@@ -12,19 +12,24 @@
 , importlib-metadata
 , substituteAll
 , runCommand
+, symlinkJoin
+, writers
+, numba
 
 , config
 
 # CUDA-only dependencies:
-, addOpenGLRunpath ? null
-, cudaPackages ? {}
+, addDriverRunpath
+, autoAddDriverRunpath ? cudaPackages.autoAddDriverRunpathHook or cudaPackages.autoAddOpenGLRunpathHook
+, cudaPackages
 
 # CUDA flags:
 , cudaSupport ? config.cudaSupport
 }:
 
 let
-  inherit (cudaPackages) cudatoolkit;
+  forBuildHost = p: p.__spliced.buildHost or p;
+  cudatoolkit = forBuildHost cudaPackages.cuda_nvcc;
```

Comment:
what does `p.__spliced.buildHost` do? why is `forBuildHost` necessary? this looks like internal, hacky stuff

looking at [the patch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/numba/cuda_path.patch), it seems like this whole thing could be simplified

Reply:
Actually, looking at the patch again, I think I picked it wrong. It seems that numba uses nvcc at runtime. I thought we needed the nvcc for the build time here

##### Broader context

In the cross-compilation setting this would select the nvcc which can be run in the build environment, instead of the one that can be run later on the host.

We don't have to directly access those when we use `nativeBuildInputs`/`buildInputs` but when we set variables in `env` or use string interpolation we have to be explicit. We're mostly ignoring this in pythonXPackages, but I think we should start making the python stuff less unfriendly to cross